src/umpire/resource/FileMemoryResource.cpp

```diff
@@ -119,10 +123,31 @@ std::size_t FileMemoryResource::getHighWatermark() const noexcept
   return 0;
 }
 
+bool FileMemoryResource::isPageable() noexcept
+{
+#if defined(UMPIRE_ENABLE_CUDA)
+  int pageableMem = 0;
+  int cdev = 0;
+  cudaGetDevice(&cdev);
+
+  //Device supports coherently accessing pageable memory
+  //without calling cudaHostRegister on it
+  cudaDeviceGetAttribute(&pageableMem,
+     cudaDevAttrPageableMemoryAccess, cdev);
+  if(pageableMem)
+    return true;
+#endif
+  // Note: Regarding omp_target, we pick a default of false here 
+  // until we can better determine which device omp_offload is using.
+  return false;
+}
+
 bool FileMemoryResource::isAccessibleFrom(Platform p) noexcept
 {
   if(p == Platform::host)
     return true;
+  else if (p == Platform::cuda) // TODO: Implement omp_target specific test
+    return isPageable();
```

Comment:
NOTE: I DO NOT think we should do this now, but one way to move the ifdef stuff out of here might be to:
```suggestion
    return CudaMemoryResource::can_access_pageable_memory();
```

Reply:
That's a good idea! I guess we can discuss this during our next meeting (and I'm about to try and summarize this into a ticket).